syntax = "proto2";

package cast_channel;

message CastMessage {
  enum PayloadType {
    STRING = 0;
    BINARY = 1;
  }
  
  optional int32 protocol_version = 1 [default = 0];
  required string source_id = 2;
  required string destination_id = 3;
  required string namespace = 4;
  required PayloadType payload_type = 5;
  optional string payload_utf8 = 6;
  optional bytes payload_binary = 7;
}